src/com/sun/jna/Memory.java

```diff
@@ -675,7 +689,7 @@ public double getDouble(long offset) {
     @Override
     public Pointer getPointer(long offset) {
         boundsCheck(offset, Native.POINTER_SIZE);
-        return super.getPointer(offset);
+        return shareIfInternalPointer(super.getPointer(offset));
```

Comment:
What do you think about `getSharedPointer` for the method name?

Reply:
It's not always a shared pointer so perhaps `conditionallyGetSharedPointer`.

Naming things [is hard](https://martinfowler.com/bliki/TwoHardThings.html).

Reply:
For APIs I think in terms of intent. How would you describe intent of this API?

I would say that we want a pointer to memory that is retained or referenced subsequently. So maybe `getReference` or `getRetainedMemoryPointer` or `getRetainedPointer` or `getSharedPointer` or `getSharablePointer`? I'm just thinking out loud.

Reply:
How about `getSharedReferenceIfInBounds`? Or since the pointer to be replaced with a shared reference is in the argument, `shareReferenceIfInBounds`. Anyway, back to really trying to test this!  Reproducing it isn't that easy.

Reply:
I'm not against discussing the color of the bikeshed, but I want to point out, that this is _not_ API. The method is `private` as it is an implementation detail of `Memory`. If the name is changed - the key point is, that the pointer, that is returned is only conditionally "shared" from the Memory object, only if it points into it.